cloudwatch_writer.go

```diff
@@ -169,6 +172,7 @@ func (c *CloudWatchWriter) queueMonitor(ctx context.Context, ticker <-chan time.
 
 	sendPayloads := func() {
 		c.sendBatch(ctx, batch, 0)
+		batchSize = 0
```

Comment:
This is the change explained in the commit message.

All the other hunks are refactoring that improves related tests - constants are exported now so they can be reused in tests. Also additional constants are introduced in tests for easier future changes.